File: templates/typescript/src/window.ts
import GObject from "gi://GObject";
import Gtk from "gi://Gtk";
import Adw from "gi://Adw";
import { registerClass } from "gjs-decorators";

@registerClass({
    GTypeName: "{{pascalCase projectName}}Window",
    Template: "resource://{{pathCase appId}}/window.ui",
    InternalChildren: ["label"],
})
export class {{pascalCase projectName}}Window extends Adw.ApplicationWindow {
    constructor(application) {
        super({ application });
    }
}

File: templates/python/{{snakeCase projectName}}/window.py
{{license}}

from gi.repository import Adw
from gi.repository import Gtk

@Gtk.Template(resource_path='{{pathCase appId}}/window.ui')
class {{pascalCase projectName}}Window(Adw.ApplicationWindow):
    __gtype_name__ = '{{pascalCase projectName}}Window'

    label = Gtk.Template.Child()

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
